### ISSUE TITLE
resource limits lose efficacy for containers in v0.17.0

### ISSUE THREAD

**Issue by njuicsgz** [GoogleCloudPlatform/kubernetes#8780]
RC configuration with limits: 
```
 resources": {
                                            "limits": {
                                                "memory": "2Gi"
                                            }
                                        }, 
```

But show unlimited in cAdvisor:
```
Memory
Limit unlimited
Swap Limit unlimited 
```

And cAdvisor show this POD used 2.37GiB memory which exceeds "2Gi" in resource limits we config. It should be killed.

**Comment by vmarmol** [comment]
What version of Docker are you running? There was an issue with the API in Docker 1.6+ which we fixed in #8491 but that was not in 0.17.0. It should be in the build this week.

**Comment by dchen1107** [comment]
I am closing this one as dup of #8814
